src/routes/profiles.py

```
@@ -1,5 +1,98 @@
+from database import UserGroupEnum, UserModel, UserProfileModel, get_db
+from exceptions import BaseS3Error, BaseSecurityError
+from schemas.profiles import RequestProfileSchema, ResponseProfileSchema
+from security.http import get_token
+from security.interfaces import JWTAuthManagerInterface
+from storages import S3StorageInterface
 
 router = APIRouter()
 
-# Write your code here
+
+@router.post(
+    "/users/{user_id}/profile/",
+    status_code=status.HTTP_201_CREATED,
+)
+async def create_user_profile(
+    user_id: int,
+    profile_data: Annotated[RequestProfileSchema, Form()],
+    db: Annotated[AsyncSession, Depends(get_db)],
+    token: Annotated[str, Depends(get_token)],
+    jwt_manager: Annotated[JWTAuthManagerInterface, Depends(get_jwt_auth_manager)],
+    s3_client: Annotated[S3StorageInterface, Depends(get_s3_storage_client)],
+) -> ResponseProfileSchema:
+    try:
+        token = jwt_manager.decode_access_token(token)
+    except BaseSecurityError as error:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail=str(error))
+    current_user = await db.scalar(
+        select(UserModel)
+        .where(UserModel.id == token.get("user_id"))
+        .options(
+            joinedload(UserModel.group),
+        )
+    )
+    if not current_user or not current_user.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="User not found or not active.",
+        )
+    if not current_user.has_group(UserGroupEnum.ADMIN) and current_user.id != user_id:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="You don't have permission to edit this profile.",
+        )
+    user_for_profile = await db.scalar(
+        select(UserModel)
+        .where(UserModel.id == user_id)
+        .options(
+            joinedload(UserModel.profile),
+        )
+    )
+    if user_for_profile.profile:
```

You access `user_for_profile.profile` without checking whether `user_for_profile` is None. If the user with `user_id` does not exist this will raise an AttributeError instead of returning the required 401 error with message "User not found or not active." Add a check such as `if not user_for_profile or not user_for_profile.is_active: raise HTTPException(status_code=401, detail="User not found or not active.")` before checking `.profile`. The task description requires this exact error behavior.